### ISSUE TITLE
Misinterpretation of sacrifices and pawn exchanges

### ISSUE THREAD

**Issue by Nazine** [pychess/pychess#945]
PyChess' current understanding of sacrifices and exchanges is still somewhat limited, as it fails to understand some peculiar types of sacrifices and pawn exchanges.

```
1. e4 e5 2. Nf3 Nf6 3. Nc3 Nc6 4. d4 exd4 5. Nxd4 Bb4 6. f3 O-O 7. Bb5 Nxd4 8.
Qxd4 c5 9. Qd3 d5 10. O-O c4 11. Qd4 Qa5 12. e5 Nd7 13. Qf2 Nxe5 14. Nxd5 Qxb5
15. Nc7 Qc6 16. Nxa8
```

(1) 15...Qc6 is a sacrifice because it lets white's knight take a rook for free. 

(2) In the same game, 9...d5 looks like a pawn exchange but PyChess instead, calls it a sacrifice.

**Comment by gbtami** [comment]
1) can't be a sacrifice because black moved hes queen out from an attack.
2) is a bug.

**Comment by gbtami** [comment]
Btw. we should comment 15. Nc7 as "fork". In general pychess should recognize other tactical motifs too.

**Comment by Nazine** [comment]
Good idea. PyChess should be able to recognize pins, forks, batteries..etc

**Comment by gbtami** [comment]
1. d4 Nf6 2. c4 e6 3. Nc3 Bb4 4. a3 Bxc3+ 5. bxc3 O-O 6. Nf3 b6 7. Bg5 h6 8. Bxf6 Qxf6 9. e3 Nc6 10. Bd3 Na5 11. O-O Bb7 12. Re1 Bxf3 13. gxf3 e5 14. dxe5 Qxe5 15. Kh1

The last move, Kh1, was clearly a sacrifice, allowing Black's queen to snatch White's pawn on c3 for nothing, but PyChess (in the comment section) doesn't say it is a sacrifice.
